clang/lib/StaticAnalyzer/Core/BasicValueFactory.cpp

```diff
@@ -275,6 +275,9 @@ BasicValueFactory::evalAPSInt(BinaryOperator::Opcode Op,
       if (V2.isSigned() && V2.isNegative())
         return nullptr;
 
+      if (V2.getBitWidth() > 64)
```

Comment:
```suggestion
      if (V2.isNegative() || V2.getBitWidth() > 64)
```
You can combine the new condition with the previous early return branch. Note that the signedness check was completely superfluous, because `APSInt::isNegative()` is defined as `isSigned() && APInt::isNegative()` [where `APInt::isNegative()` checks the sign bit].

Reply:
Will do, thanks